### ISSUE TITLE
[Feature request] Add Anilist

### ISSUE THREAD

**Issue by TronTheGamer** [LNReader/lnreader#42]

**Comment by Zaroz** [comment]
Anilist is for anime and manga. I don't think it works with novels

**Comment by HyphenSam** [comment]
> Anilist is for anime and manga. I don't think it works with novels

[AniList does track light novels.](https://anilist.co/search/manga?format=NOVEL)

**Comment by shipoxx** [comment]
> Anilist is for anime and manga. I don't think it works with novels

No it's actually support novel now

**Comment by stepan323446** [comment]
I support it. I would really like Anilist.

**Comment by BicheT0rdue** [comment]
yes please, i need this feature 🙏

**Comment by 1-minute-to-midnight** [comment]
+1 on this one

**Comment by TehNut** [comment]
I'm interested in working on this, but I've never successfully gotten an Android development environment set up, so it may be a bit before I can work on it 😅

After a bit of digging around, it doesn't seem like the *most* straightforward thing to add new tracker support. I may be missing something (it was just a quick skim over GitHub) but it looks very hardcoded to specifically use MAL. If I were to look into this, there's 2 routes I could go:

1. A similar, somewhat hardcoded approach. There wouldn't be many large changes, just additions. Any future trackers would have to go through it as well.
2. Reworking trackers into something easily extendible. Would include more sweeping changes, but adding additional trackers in the future would be simpler.

Which route would be preferred by the team if I were to move forward with this?

**Comment by TehNut** [comment]
Well, I decided to take a stab at it with option 1. (https://github.com/TehNut/lnreader/commit/829ff3f7f540db2d8942e6fb8841430b532bc0e9) 

As mentioned in the commit message, there's a large data model change. This was done to provide a normalized interface for different trackers, making it simpler to add them in the future. I haven't done plain JS in a long time, so I wasn't sure how to "enforce" a shape for the trackers, hence the really dumb looking `createTracker` function. Not much else uses TS, so I tried to stay away from it.

https://user-images.githubusercontent.com/5789439/236612339-5d7ce1f3-aaf4-4c0d-8bb5-17f632117fe9.mp4

**Comment by TehNut** [comment]
I think I've covered everything, honestly 🤔 I don't know React so I'm sure there's some stuff that could be cleaned up. I generalized some of the components because so few changes were needed between trackers.

All that's really left is looking at migrating old tracker data to the new format. Not really sure where to start with that, though.

https://github.com/LNReader/lnreader/compare/main...TehNut:lnreader:tracker/anilist

https://user-images.githubusercontent.com/5789439/236643626-92a4cd66-6b53-45e6-aa5f-51d9d6ab96c5.mp4
